app/controllers/pages_controller.rb

```diff
@@ -54,7 +54,14 @@ def algorithme
   end
 
   def faq
-    @faq_items = FaqItem.all
+    respond_to do |format|
+      format.html do
+        @faq_items = Rails.cache.fetch("faq_items", expires_in: 2.hours) { FaqItem.all }
+      end
+      format.json do
+        render json: Rails.cache.fetch("faq_items_json", expires_in: 2.hours) { FaqItem.all.to_json }
+      end
+    end
```

Comment:
Y a t'il une raison spéciale pour avoir deux clés de cache ?
Sinon, cette simplification ferait-elle l'affaire ?

```suggestion
    @faq_items = Rails.cache.fetch("faq_items", expires_in: 2.hours) { FaqItem.all }
    respond_to do |format|
      format.html
      format.json { render json: @faq_items.to_json }
    end
```

Reply:
Oui je pense que tu as raison, faire deux caches c'est plus couteux que de faire un cache et un call ruby .to_json